src/main/scala/de/unruh/isabelle/pure/Transition.scala

```diff
@@ -0,0 +1,109 @@
+package de.unruh.isabelle.pure
+
+import de.unruh.isabelle.control.Isabelle
+import de.unruh.isabelle.mlvalue.Implicits.listConverter
+import de.unruh.isabelle.mlvalue.{MLValue, MLValueWrapper}
+import Transition.Ops
+
+import scala.language.postfixOps
+
+// Implicits
+import de.unruh.isabelle.mlvalue.Implicits._
+import de.unruh.isabelle.pure.Implicits.theoryConverter
+
+/** Represents a transition between proof states (ML type `Transition`) in the Isabelle process.
+  *
+  * Use `Transition.parseOuterSyntax` to parse Isar source code into into a list of transitions.
+  *
+  * An instance of this class is merely a thin wrapper around an [[mlvalue.MLValue MLValue]].
+  */
+final class Transition private[Transition] (val mlValue: MLValue[Transition])
+    extends MLValueWrapper[Transition] {
+
+  /** Returns the command name (ML function `Toplevel.name_of`).
+   *
+   * Examples: "theory", "section", "lemma", "by", "using", "definition", 'locale", "record", "end".
+   * The name is "<ignored>" if and only if isIgnored is true.
+   * The name is "<malformed>" if and only if isMalformed is true.
+   */
+  def name(implicit isabelle: Isabelle): String = Ops.getName(this).retrieveNow
+  /** Returns the position of this transition's command name. */
+  def position(implicit isabelle: Isabelle): Position = Ops.getPosition(this).retrieveNow
+
+  /** Returns whether this transition introduces a theory ("theory .. begin", ML function `Toplevel.is_init`).
+   *
+   * Init transitions can only be applied to toplevel states in theory mode.
+   */
+  def isInit(implicit isabelle: Isabelle): Boolean = Ops.isInit(this).retrieveNow
+  /** Returns whether the transition is just whitespace, comments, text (ML function `Toplevel.is_ignored`). */
+  def isIgnored(implicit isabelle: Isabelle): Boolean = Ops.isIgnored(this).retrieveNow
+  /** Returns true if the transition is a parsing error  (ML function `Toplevel.is_malformed`). */
+  def isMalformed(implicit isabelle: Isabelle): Boolean = Ops.isMalformed(this).retrieveNow
+
+  /** Execute the transition on a TopLevelState (ML function `Toplevel.command_exception`).
+   *
+   * TODO What does the `int` parameter do? Probably only affects internal Isabelle 'presentation'.
```

Comment:
I've seen `int` in other places, too. (Somewhere in Skip_Proof for example it makes a big difference, if I remember right.) I believe it means "interactive", i.e., whether this command is entered in an interactive theory (e.g., opened in jEdit), or in a batch (e.g., through `isabelle build`).

Reply:
Ok, sounds reasonable, I updated the docstring. (I tried tracing what it does through ML code but it's just impossible.)

Reply:
I think(!) mostly it's just passed through everywhere (it occurs in many different places), and some very few places actually use it. For example, if you have a `sorry` without `quick_and_dirty` configuration option, then it will raise an error when `int=false`. Meaning you can always use `sorry` in jEdit. But not in the build.